threads/src/main/kotlin/pl/edu/pw/ia/threads/domain/query/projector/ThreadProjector.kt

```diff
@@ -52,6 +58,27 @@ class ThreadProjector(
 		repository.delete(event.threadId)
 	}
 
+	@EventHandler
+	fun on(event: AccountUpdatedEvent) {
+		repository.findByAccountId(event.accountId)
+			.map { it.copy(username = event.name) }
+			.subscribe {repository.save(it)}
+	}
+
+	@EventHandler
+	fun on(event: PostCreatedEvent) {
+		val view = repository.findByIdAndPostIdIsNull(event.threadId) ?: throw ThreadNotFoundException(event.threadId)
```

Comment:
here, we will throw an exception every time post is created under thread with at leas one post